### PR TITLE
[tempo-distributed] fix-ingress-query-frontend-typo

### PR DESCRIPTION
**Pull request by chrisduong**
Signed-off-by: Chris Duong <chrisduong83@gmail.com>

### CODE REVIEW

File: charts/tempo-distributed/templates/_helpers.tpl [hunk 1]
```diff
@@ -152,7 +152,7 @@ Return if ingress supports pathType.
 Return the appropriate apiVersion for PodDisruptionBudget.
 */}}
 {{- define "tempo.pdb.apiVersion" -}}
-  {{- if .Capabilities.APIVersions.Has "policy/v1/PodDisruptionBudget" -}}
+  {{- if .Capabilities.APIVersions.Has "policy/v1" -}}
```
Comment (zanhsieh) [inline charts/tempo-distributed/templates/_helpers.tpl:155]: Please revert the change.

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/tempo-distributed/templates/_helpers.tpl (hunk 1 above)
Comment (chrisduong) [reply charts/tempo-distributed/templates/_helpers.tpl:155]: Hi @zanhsieh, could we support ArgoCD here. ArgoCD run `helm template`, not `helm install`, and it only passes `--api-versions="policy/v1", not "policy/v1/PodDisruptionBudget". It wouldn't break anything.

It is also consistent with this match ("autoscaling/v2") https://github.com/grafana/helm-charts/blob/0f044cce08476523d27f80913696bb25e5977ca4/charts/tempo-distributed/templates/_helpers.tpl#L166 

```
❯ kubeclt api-versions | grep policy
policy/v1
policy/v1beta1
```

Comment (zanhsieh) [reply charts/tempo-distributed/templates/_helpers.tpl:155]: > Hi @zanhsieh, could we support ArgoCD here. ArgoCD run `helm template`, not `helm install`, and it only passes `--api-versions="policy/v1"`, not "policy/v1/PodDisruptionBudget". It wouldn't break anything.
> 
> It is also consistent with this match ("autoscaling/v2")
> 
> https://github.com/grafana/helm-charts/blob/0f044cce08476523d27f80913696bb25e5977ca4/charts/tempo-distributed/templates/_helpers.tpl#L166
> 
> ```
> ❯ kubeclt api-versions | grep policy
> policy/v1
> policy/v1beta1
> ```

No. Your topic is `[[tempo-distributed] fix-ingress-query-frontend-typo](https://github.com/grafana/helm-charts/pull/2421#top)` which is nothing related to ArgoCD.

Comment (chrisduong) [reply charts/tempo-distributed/templates/_helpers.tpl:155]: Hi @zanhsieh, I had printed the `.Capabilities.APIVersions`, it does not contains "policy/v1/PodDisruptionBudget".

```gotmpl
{{ toPrettyJson .Capabilities.APIVersions }}
```

Output:

```
[
  "v1",
  "admissionregistration.k8s.io/v1",
  "admissionregistration.k8s.io/v1alpha1",
  "admissionregistration.k8s.io/v1beta1",
  "internal.apiserver.k8s.io/v1alpha1",
  "apps/v1",
  "apps/v1beta1",
  "apps/v1beta2",
  "authentication.k8s.io/v1",
  "authentication.k8s.io/v1alpha1",
  "authentication.k8s.io/v1beta1",
  "authorization.k8s.io/v1",
  "authorization.k8s.io/v1beta1",
  "autoscaling/v1",
  "autoscaling/v2",
  "autoscaling/v2beta1",
  "autoscaling/v2beta2",
  "batch/v1",
  "batch/v1beta1",
  "certificates.k8s.io/v1",
  "certificates.k8s.io/v1beta1",
  "coordination.k8s.io/v1beta1",
  "coordination.k8s.io/v1",
  "discovery.k8s.io/v1",
  "discovery.k8s.io/v1beta1",
  "events.k8s.io/v1",
  "events.k8s.io/v1beta1",
  "extensions/v1beta1",
  "flowcontrol.apiserver.k8s.io/v1alpha1",
  "flowcontrol.apiserver.k8s.io/v1beta1",
  "flowcontrol.apiserver.k8s.io/v1beta2",
  "flowcontrol.apiserver.k8s.io/v1beta3",
  "networking.k8s.io/v1",
  "networking.k8s.io/v1alpha1",
  "networking.k8s.io/v1beta1",
  "node.k8s.io/v1",
  "node.k8s.io/v1alpha1",
  "node.k8s.io/v1beta1",
  "policy/v1",
  "policy/v1beta1",
  "rbac.authorization.k8s.io/v1",
  "rbac.authorization.k8s.io/v1beta1",
  "rbac.authorization.k8s.io/v1alpha1",
  "resource.k8s.io/v1alpha1",
  "scheduling.k8s.io/v1alpha1",
  "scheduling.k8s.io/v1beta1",
  "scheduling.k8s.io/v1",
  "storage.k8s.io/v1beta1",
  "storage.k8s.io/v1",
  "storage.k8s.io/v1alpha1",
  "apiextensions.k8s.io/v1beta1",
  "apiextensions.k8s.io/v1"
]
```

So the right code is

```gotmpl
{{- if .Capabilities.APIVersions.Has "policy/v1" -}}
```

Comment (zanhsieh) [reply charts/tempo-distributed/templates/_helpers.tpl:155]: > Hi @zanhsieh, I had printed the `.Capabilities.APIVersions`, it does not contains "policy/v1/PodDisruptionBudget".
> 
> ```
> {{ toPrettyJson .Capabilities.APIVersions }}
> ```
> 
> Output:
> 
> ```
> [
>   "v1",
>   "admissionregistration.k8s.io/v1",
>   "admissionregistration.k8s.io/v1alpha1",
>   "admissionregistration.k8s.io/v1beta1",
>   "internal.apiserver.k8s.io/v1alpha1",
>   "apps/v1",
>   "apps/v1beta1",
>   "apps/v1beta2",
>   "authentication.k8s.io/v1",
>   "authentication.k8s.io/v1alpha1",
>   "authentication.k8s.io/v1beta1",
>   "authorization.k8s.io/v1",
>   "authorization.k8s.io/v1beta1",
>   "autoscaling/v1",
>   "autoscaling/v2",
>   "autoscaling/v2beta1",
>   "autoscaling/v2beta2",
>   "batch/v1",
>   "batch/v1beta1",
>   "certificates.k8s.io/v1",
>   "certificates.k8s.io/v1beta1",
>   "coordination.k8s.io/v1beta1",
>   "coordination.k8s.io/v1",
>   "discovery.k8s.io/v1",
>   "discovery.k8s.io/v1beta1",
>   "events.k8s.io/v1",
>   "events.k8s.io/v1beta1",
>   "extensions/v1beta1",
>   "flowcontrol.apiserver.k8s.io/v1alpha1",
>   "flowcontrol.apiserver.k8s.io/v1beta1",
>   "flowcontrol.apiserver.k8s.io/v1beta2",
>   "flowcontrol.apiserver.k8s.io/v1beta3",
>   "networking.k8s.io/v1",
>   "networking.k8s.io/v1alpha1",
>   "networking.k8s.io/v1beta1",
>   "node.k8s.io/v1",
>   "node.k8s.io/v1alpha1",
>   "node.k8s.io/v1beta1",
>   "policy/v1",
>   "policy/v1beta1",
>   "rbac.authorization.k8s.io/v1",
>   "rbac.authorization.k8s.io/v1beta1",
>   "rbac.authorization.k8s.io/v1alpha1",
>   "resource.k8s.io/v1alpha1",
>   "scheduling.k8s.io/v1alpha1",
>   "scheduling.k8s.io/v1beta1",
>   "scheduling.k8s.io/v1",
>   "storage.k8s.io/v1beta1",
>   "storage.k8s.io/v1",
>   "storage.k8s.io/v1alpha1",
>   "apiextensions.k8s.io/v1beta1",
>   "apiextensions.k8s.io/v1"
> ]
> ```
> 
> So the right code is
> 
> ```
> {{- if .Capabilities.APIVersions.Has "policy/v1" -}}
> ```

No, you should do this:

```console
$ kubectl api-resources | grep -i budget
poddisruptionbudgets              pdb          policy/v1                              true         PodDisruptionBudget

$ k version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.9", GitCommit:"a1a87a0a2bcd605820920c6b0e618a8ab7d117d4", GitTreeState:"clean", BuildDate:"2023-04-12T12:16:51Z", GoVersion:"go1.19.8", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.3", GitCommit:"434bfd82814af038ad94d62ebe59b133fcb50506", GitTreeState:"clean", BuildDate:"2022-10-25T19:35:11Z", GoVersion:"go1.19.2", Compiler:"gc", Platform:"linux/amd64"}
```

Comment (chrisduong) [reply charts/tempo-distributed/templates/_helpers.tpl:155]: I digged a little big deeper, it might be caused by how ArgoCD used "managedField" when it is Server-Side Apply mode.

I reverted this change.

**Review by zanhsieh — APPROVED** [review]
